### PR TITLE
feat: add loggers package

### PR DESCRIPTION
**Pull request by emmenko**
Partially part of #1444 

This PR open sources our internal loggers package, used by our backend services.

### CODE REVIEW

File: packages-backend/loggers/src/custom-formats/json.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import unsetIn from 'lodash/unset';
+
+/*
+ * function replacer (key, value)
+ * Handles proper stringification of Buffer output.
+ */
+function replacer(key: string, value: Buffer | string) {
+  if (key === 'queryJsonString') {
+    // We need to stringify the value as otherwise Kibana cardinality explodes.
+    // Ref: https://commercetools.slack.com/archives/C011FLZ1JLW/p1588066334005900
```
Comment (tdeekens) [inline packages-backend/loggers/src/custom-formats/json.ts:19]: I think we should remove references from internal Slack convos.

File: packages-backend/loggers/src/create-error-report-logger.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    DSN: string;
+    role: string;
+    environment: string;
+  };
+  errorMessageBlacklist?: Array<string | RegExp>;
+};
+
+function createErrorReportLogger(options: LoggerOptions = {}) {
+  // Sentry
+  if (options.sentry) {
```
Comment (tdeekens) [inline packages-backend/loggers/src/create-error-report-logger.ts:20]: Hm. Personally I find it rather specific that we use Sentry. Do we know if others using this package would too or should we allow for more reporting targets?

Comment (emmenko) [reply packages-backend/loggers/src/create-error-report-logger.ts:20]: In my defense, the package says "_Opinionated JSON loggers_" 😜

**Comment by emmenko** [comment]
I understand your points, let's have a chat then 🙂

File: packages-backend/loggers/src/custom-formats/json.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * object into pure JSON. This was previously exposed as { json: true }
+ * to transports in `winston < 3.0.0`.
+ */
+/* eslint-disable no-param-reassign */
+const jsonFormatter = format((info, opts = {}) => {
+  info.logLevel = info.level;
+  delete info.level;
+
+  // Replace / rename fields
+  replaceField(info, 'meta.req.query', 'meta.req.queryJsonString');
```
Comment (tdeekens) [inline packages-backend/loggers/src/custom-formats/json.ts:48]: Should be an option `omitMetaPaths: String[]` I think.

File: packages-backend/loggers/src/create-access-logger.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+// Inspired by https://github.com/bithavoc/express-winston/issues/62#issuecomment-396906056
+const requestFilter: RequestFilter = (req, propName) => {
+  if (propName === 'headers') {
+    return redactInsecureRequestHeaders(req);
+  }
+  return req[propName];
+};
+
+const createAccessLogger = (options: LoggerOptions = {}) => {
```
Comment (tdeekens) [inline packages-backend/loggers/src/create-access-logger.ts:26]: This is a middleware I think. So to have it inline with others maybe it's `createAccessLoggerMiddleware(options)`.

File: packages-backend/loggers/src/env.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const env = process.env.NODE_ENV;
+const isDev = !env || env === 'development';
+const isProd = env === 'production';
+const isTest = env === 'test';
+
+export { isDev, isProd, isTest };
```
Comment (tdeekens) [inline packages-backend/loggers/src/env.ts:6]: I would also have this as an arg to the middlewars. `envGetters` maybe. Not 100% also just making notes for myself.

File: packages-backend/loggers/src/custom-formats/json.ts (hunk 3 above)
Comment (emmenko) [reply packages-backend/loggers/src/custom-formats/json.ts:48]: Yeah, probably we need to provide some kind of fields mapping.

```js
{
  rewriteMetaInfoFields: {
    'meta.req.query': 'meta.req.queryJsonString'
  }
}
```

An alternative might be to allow to pass a custom `logform` formatter. We can also expose a bunch of default formatter (e.g. json, kibana, etc). The kibana formatter can also be additionally configured with the fields rewriter.

File: packages-backend/loggers/src/create-error-report-logger.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import type { Request, Response, NextFunction } from 'express';
+
+import * as Sentry from '@sentry/node';
+import * as env from './env';
+import redactInsecureRequestHeaders from './utils/redact-insecure-request-headers';
```
[the remark this answers is not in the document]
Comment (emmenko) [reply packages-backend/loggers/src/create-error-report-logger.ts:5]: PS: one of the intents with this loggers package was to provide preconfigured loggers, meaning that for common cases there should be no need to provide extra configuration.

We could maybe have a default list of "insecure" headers, while still allowing to extend that list.

File: packages-backend/loggers/src/custom-formats/json.ts (hunk 3 above)
Comment (tdeekens) [reply packages-backend/loggers/src/custom-formats/json.ts:48]: Yeah. I think sane defaults which can be overwritten would be good. Maybe also by just saying: I want a Kibana logger as a type argument you get those defaults.

**Comment by emmenko** [comment]
I cleaned things up and remove all unnecessary things. I also removed Sentry, as I think we can keep this internal for now.

@tdeekens can you take another look?

File: packages-backend/loggers/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+// Loggers
+export { default as createApplicationLogger } from './create-application-logger';
+
+// Formatters
+export { default as rewriteFieldsFormatter } from './formatters/rewrite-fields';
+
+// Re-export winston for convenience
+import * as winston from 'winston';
+export { winston };
```
Comment (tdeekens) [inline packages-backend/loggers/src/index.ts:12]: Nit: doesn't TS allow `export * as winston` from 'winston'`?

**Pull request merged**